### PR TITLE
Login

### PR DESCRIPTION
**Pull request by bloodbare**

### CODE REVIEW

File: projects/grange/src/lib/views/login.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
+import { Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
 import { Grange } from '../grange.service';
 import { LoginInfo } from 'grange-core';
+import { markForCheck } from 'develop/pastanaga-angular/projects/pastanaga/src/lib/common/utils';
```
Comment (ebrehault) [inline projects/grange/src/lib/views/login.ts]: your IDE took the wrong path:
```
import { markForCheck } from 'pastanaga-angular`;
```

File: projects/grange/src/lib/views/forgot.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
+import { Grange } from '../grange.service';
+import { RecoverInfo } from 'grange-core';
+import { markForCheck } from 'develop/pastanaga-angular/projects/pastanaga/src/lib/common/utils';
```
Comment (ebrehault) [inline projects/grange/src/lib/views/forgot.ts]: wrong path

File: projects/grange/src/lib/views/login.ts [hunk 3]
```diff
@@ -1,60 +1,78 @@
...
-    useToken = false;
-    error = '';
-    isLogged: boolean;
-    allowRegister: boolean;
-    allowSocial: boolean;
-    allowForgot: boolean;
-    allowToken: boolean;
+  useToken = false;
+  error = '';
+  isLogged: boolean;
```
Comment (ebrehault) [inline projects/grange/src/lib/views/login.ts]: we need to respect strict compilation, so an attribute must either defined directly (here or in constructor), either marked explicitly as possibly undefined:
```
title?: string;
```

File: projects/grange/src/lib/views/login.ts [hunk 4]
```diff
@@ -1,60 +1,78 @@
...
+  title: string;
+  allowRegister: boolean;
```
Comment (ebrehault) [inline projects/grange/src/lib/views/login.ts]: all booleans should be assigne to `false` by default (unless `undefined` is a possible meaningful state)

File: projects/grange/src/lib/views/login.ts [hunk 5]
```diff
@@ -1,60 +1,78 @@
...
+  onForgot() {
+    this.grange.traverser.traverse('/@@forgot');
+  }
 
-    onGoogleLogin() {
-      // We need to call
-      // {{API_URL}}/@authenticate/google?callback={{WEB_URL}}/@@callback/google"
-    }
+  socialLogin(provider: string) {
+    const callUrl = this.config.get('BACKEND_URL') + '/@authenticate/' + provider + '?callback={{WEB_URL}}/@@callback/' + provider;
```
Comment (ebrehault) [inline projects/grange/src/lib/views/login.ts]: it should be done in a grange-core service

File: projects/grange/src/lib/views/validation.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  ngOnInit() {
+
+    this.grange.core.auth.isAuthenticated.subscribe(auth => {
+      this.isLogged = auth.state;
+      markForCheck(this.cdr);
+    });
+    debugger;
+    this.token = '';
+    this.grange.core.api.get('/@validate_schema/' + this.token).subscribe(
```
Comment (ebrehault) [inline projects/grange/src/lib/views/validation.ts]: better to have a method in grange-core doing the endpoint call

File: projects/grange/src/lib/views/validation.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      res => {
+        this.schema = res;
+        markForCheck(this.cdr);
+      }
+    );
+  }
+
+  onSubmit(data: RecoverInfo) {
+    this.error = '';
+    this.grange.core.api.post('/@validation/' + this.token, this.editModel).subscribe(
```
Comment (ebrehault) [inline projects/grange/src/lib/views/validation.ts]: same

File: mrs.developer.json [hunk 8]
```diff
@@ -3,36 +3,36 @@
...
         "https": "https://github.com/guillotinaweb/angular-traversal.git",
-        "branch": "master"
+        "tag": "1.2.6"
     },
     "grange-core": {
         "url": "git@github.com:guillotinaweb/grange-core.git",
         "https": "https://github.com/guillotinaweb/grange-core.git",
         "path": "/projects/grange-core/src",
-        "branch": "master"
+        "branch": "wip_login"
```
Comment (ebrehault) [inline mrs.developer.json]: put the proper grange-core version once the other PR is merged

File: projects/grange/src/lib/views/user/forgot.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  ngOnInit() {
+    this.grange.core.auth.isAuthenticated.subscribe(auth => {
+      this.isLogged = auth.state;
+      markForCheck(this.cdr);
+    });
+  }
+
+  onSubmit(data: RecoverInfo) {
+    this.error = '';
+    this.grange.core.api.post('/@users/' + data.login + '/reset-password', {}).subscribe(
```
Comment (ebrehault) [inline projects/grange/src/lib/views/user/forgot.ts]: you should use `requestPasswordReset` instead:
```
this.grange.core.auth.requestPasswordReset()
```

File: mrs.developer.json (hunk 8 above)
Comment (bloodbare) [reply mrs.developer.json]: Its still WIP, so yes

**Pull request merged**